src/data-workspace/category-combo-table-body/category-combo-table-body.test.js

```diff
@@ -0,0 +1,770 @@
+                code: 'COC_291',
+                displayName: 'Fixed, >1y',
+                id: 'psbwp3CQEhs',
+                name: 'Fixed, >1y',
+            },
+        ],
+        created: '2011-12-24T12:24:25.203',
+        dataDimensionType: 'DISAGGREGATION',
+        displayName: 'Location and age group',
+        favorite: false,
+        isDefault: false,
+        lastUpdated: '2011-12-24T12:24:25.203',
+        name: 'Location and age group',
+    },
+    bjDvmb4bfuf: {
+        categories: ['GLevLNI9wkl'],
+        categoryOptionCombos: [
+            {
+                code: 'default',
+                name: 'default',
+                categoryOptions: ['xYerKDKCefk'],
+                displayName: 'default',
+                id: 'HllvX50cXC0',
+            },
+        ],
+        code: 'default',
+        created: '2011-12-24T12:24:25.203',
+        dataDimensionType: 'DISAGGREGATION',
+        displayName: 'default',
+        favorite: false,
+        id: 'bjDvmb4bfuf',
+        isDefault: true,
+        lastUpdated: '2016-04-12T20:37:49.126',
+        name: 'default',
+    },
+}
+
+const metadata = {
+    categories,
+    categoryOptions,
+    categoryCombos,
+    dataElements,
+}
+
+describe('<CategoryComboTableBody />', () => {
+    useMetadata.mockReturnValue({ data: metadata })
+
+    it('should render 15 input rows and 5 columns per input', () => {
+        const tableDataElements = [
+            dataElements.s46m5MS0hxu,
+            dataElements.UOlfIjgN8X6,
+            dataElements.z7duEFcpApd,
+            dataElements.x3Do5e7g4Qo,
+            dataElements.pikOziyCXbM,
+            dataElements.O05mAByOgAv,
+            dataElements.vI2csg55S9C,
+            dataElements.xc8gmAKfO95,
+            dataElements.mGN1az8Xub6,
+            dataElements.L2kxa2IA2cs,
+            dataElements.fClA2Erf6IO,
+            dataElements.I78gJm4KBo7,
+            dataElements.n6aMJNLdvep,
+            dataElements.l6byfWFUGaP,
+            dataElements.YtbsuPPo010,
+        ]
+
+        const result = render(
+            <Table>
+                <CategoryComboTableBody
+                    categoryCombo={categoryCombos.dzjKKQq0cSO}
+                    dataElements={tableDataElements}
+                />
+            </Table>,
+            {
+                wrapper: ({ children }) => (
+                    <FinalFormWrapper dataValueSet={{}}>
+                        {children}
+                    </FinalFormWrapper>
+                ),
+            }
+        )
+
+        const inputRows = result.getAllByTestId('dhis2-dataentry-tableinputrow')
+        expect(inputRows.length).toBe(15)
+
+        inputRows.forEach((inputRow, index) => {
+            const dataElement = tableDataElements[index]
+            const headerCell = getByText(inputRow, dataElement.displayName, {
+                selector: '[data-test="dhis2-dataentryapp-dataelementcell"]',
+            })
+
+            const inputCells = getAllByTestId(
```

Comment:
Should we also check that these render actual input fields with `dataElementId.cocId` as the name?

Reply:
That seems a little tricky. In that case I think it'd be better to check that a `<DataEntryField />` has been rendered. Whether an input has the correct name should be a different test. In this component a field could be an input, several inputs (radio buttons), a button or a textbox. That makes it quite hard to test it properly